docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+
+Given a list of magnet links provided by the index to download individual archives, member nodes can:
+
+1. **Download all archives** - Extract each magnet link in the index and pass them to the underlying BitTorrent client (this is the case for new community member nodes that haven't downloaded any archives yet)
+2. **Download only the latest archive** - Extract only the newest magnet link and pass it to the BitTorrent client (this the case for any member node that already has downloaded all previous history and is now interested in only the latst archive)
+3. **Download specific archives** - Look into `from` and `to` fields of every `WakuMessageArchiveMetadata` and only extract magnet links for archives of a specific time range (can be the case for member nodes that have recently joined the network and are only interested in a subset of the complete history)
+
+
+## Storing historical messages
+
+When message archives are fetched, community member nodes MUST unwrap the resulting `WakuMessage` instances into `ApplicationMetadataMessage` instances and store them in their local database.
+Community member nodes SHOULD NOT store the wrapped `WakuMessage` messages.
+
+Already stored messages with the same `id` or `clock` value MUST be replaced with messages extracted from archives, if both of these values are equal.
+
+Community members nodes MUST ignore the expiration state of each archive message.
+
+## Considerations
+
+The following are things to cosider when implementing this specification.
+
+### Bandwidth consumption
+
+Community member nodes will download the latest archive they've received from the archive index, which includes messages from the last seven days. Assuming that community members nodes were online for that time range, they have already downloaded that message data and will now download an archive that contains the same.
+
+This means there's a possibility member nodes will download the same data at least twice.
+
+### Multiple community owners
+
+It is possible for community owners to export the private key of their owned community and pass it to other users so they become community owners as well. This means, it's possible for multiple owners to exist.
+
+This might conflict with the assumption that the community owner node serves as a single source of thruth. Multiple owners can have different message histories.
+
+Not only will multiple owners multiply the amount of archive index messages being distributed to the network, they might also contain different sets of magnet links and their corresponding hashes.
+
+Even if just a single message is missing in one of the histories, the hashes presented in archive indices will look completely different, resulting in the community member node to download the corresponding archive (which might be identical to an archive that was already downloaded, except for that one message).
```

Comment:
@John-44 very important consideration that I think we haven't talked about yet.

Reply:
@PascalPrecht yes it is, well spotted!

As soon as we have tokenised community ownership it won't be possible for two people (aka ethereum accounts) to own a single community, but there is nothing to stop a single person signing into two Status Desktop instances with the same profile and therefore their same account (that owns a community) *would* be running on two nodes.

Could we detect if this occurs, and automatically select only one node out of however many nodes the owner has spun up to be the node that produces history?  We could also expose this setting to the community owner, to let the community owner select a different node to be the node that produces history perhaps?

What do you think?

Reply:
> single message is missing in one of the histories, the hashes presented in archive indices

Agree, and if we can make design robust to this it'd be useful. I suppose this is related to the whole archival index discussion? (I haven't kept up here in detail, just noticed a lot of back and forth).

Reply:
> We could also expose this setting to the community owner, to let the community owner select a different node to be the node that produces history perhaps?

Yes, I think what we can do is set a "main owner". So even if there are multiple ppl with private keys, only one main owner could be set. Obviously, with multiple owners having the private key and write privileges, each of them can change that value as they like.

This could still be problematic if they serve different archives. Then the question is: will member nodes simply ignore all the older archives in a given time range (because they might look completely different), or will they also download all of it and keep replacing all of it.

In other words: If member nodes detect that the history has changed, will they replace that entire history, or will they stick to only downloading the latest #n archives?